src/Interactive/EditorFeatures/Core/Extensibility/Interactive/NuGetPackageResolverImpl.cs

```diff
@@ -0,0 +1,180 @@
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+
+namespace Microsoft.CodeAnalysis.Editor.Interactive
+{
+    internal sealed class NuGetPackageResolverImpl : WORKSPACES::Microsoft.CodeAnalysis.Scripting.NuGetPackageResolver
+    {
+        private readonly string _packagesDirectory;
+
+        internal NuGetPackageResolverImpl(string packagesDirectory)
+        {
+            Debug.Assert(PathUtilities.IsAbsolute(packagesDirectory));
+            _packagesDirectory = packagesDirectory;
+        }
+
+        internal override ImmutableArray<string> ResolveNuGetPackage(string reference)
+        {
+            string packageName;
+            string packageVersion;
+            if (!ParsePackageReference(reference, out packageName, out packageVersion))
+            {
+                return default(ImmutableArray<string>);
+            }
+
+            try
+            {
+                var tempPath = PathUtilities.CombineAbsoluteAndRelativePaths(Path.GetTempPath(), Guid.NewGuid().ToString("D"));
+                var tempDir = Directory.CreateDirectory(tempPath);
+                try
+                {
+                    // Create project.json.
+                    var projectJson = PathUtilities.CombineAbsoluteAndRelativePaths(tempPath, "project.json");
+                    using (var stream = File.OpenWrite(projectJson))
+                    using (var writer = new StreamWriter(stream))
+                    {
+                        WriteProjectJson(writer, packageName, packageVersion);
+                    }
+
+                    // Run "nuget.exe restore project.json" to generate project.lock.json.
+                    NuGetRestore(projectJson);
+
+                    // Read the references from project.lock.json.
+                    var projectLockJson = PathUtilities.CombineAbsoluteAndRelativePaths(tempPath, "project.lock.json");
+                    using (var stream = File.OpenRead(projectLockJson))
+                    using (var reader = new StreamReader(stream))
+                    {
+                        return ReadProjectLockJson(_packagesDirectory, reader);
+                    }
+                }
+                finally
+                {
+                    tempDir.Delete(recursive: true);
+                }
+            }
+            catch (IOException)
+            {
+            }
+            catch (UnauthorizedAccessException)
+            {
+            }
+            return default(ImmutableArray<string>);
+        }
+
+        /// <summary>
+        /// Syntax is "id/version", matching references in project.lock.json.
+        /// </summary>
+        internal static bool ParsePackageReference(string reference, out string name, out string version)
+        {
+            int separator = reference.IndexOf('/');
+            int n = reference.Length;
+            if ((separator > 0) &&
+                (separator < n - 1) &&
+                (reference.IndexOf('/', separator + 1) < 0))
```

Comment:
string.Split with max size == 2 might just be easier here. I don't think a single array allocation will kill us here.